applications/serial_lte_modem/src/lwm2m_carrier/slm_at_carrier.c

```diff
@@ -99,11 +103,30 @@ static struct carrier_op_list op_list[CARRIER_OP_MAX] = {
 
 #define SLM_CARRIER_OP_STR_MAX (sizeof("battery_status") + 1)
 
-/* Global variables and functions defined in different files. */
+struct k_work_delayable reconnect_work;
+
+/* Global variables defined in different files. */
+extern int32_t auto_connect;
 extern struct k_work_q slm_work_q;
 extern struct at_param_list at_param_list;
 extern uint8_t data_buf[SLM_MAX_MESSAGE_SIZE];
 
+/* Global functions defined in different files. */
+int slm_setting_fota_save(void);
+int nidd_auto_connect(void);
+
+NRF_MODEM_LIB_ON_INIT(lwm2m_init_hook, on_modem_lib_init, NULL);
+
+static void on_modem_lib_init(int ret, void *ctx)
+{
+	ARG_UNUSED(ctx);
+
+	/** ret: Zero on success, a positive value @em nrf_modem_dfu when executing
+	 *  Modem firmware updates, and negative errno on other failures.
+	 */
+	LOG_INF("lib_modem init: %d", ret);
```

Comment:
Perhaps this hook could be made generic for the SLM? Don't think it has much to do with the Carrier library per se, the library has its own hook in `lwm2m_carrier.c`, and the modem initialization in SLM takes place in main.

Reply:
yes, good idea as SLM need the log for fault handling as well.